R/utils-cut.R

```diff
@@ -44,12 +44,25 @@ cut_width <- function(x, width, center = NULL, boundary = NULL, closed = c("righ
 
   # Determine bins
   min_x <- find_origin(x_range, width, boundary)
-  # Small correction factor so that we don't get an extra bin when, for
-  # example, origin = 0, max(x) = 20, width = 10.
-  max_x <- max(x, na.rm = TRUE) + (1 - 1e-08) * width
+  max_x <- max(x, na.rm = TRUE)
 
-  breaks <- c(seq(min_x, max_x, width), max_x)
-  cut(x, breaks, include.lowest = TRUE, right = (closed == "right"), ...)
+  breaks <- c(seq(min_x, max_x, width))
+  # Round breaks *before* they go into the cut function. This way the data (not rounded)
+  # will be correctly divided into the rounded bins
+  if (all(x %% 1 == 0)) {
+    avgDigits <- 0
+  } else {
+    avgDigits <- floor(mean(stringi::stri_count_regex(as.character(x), "[[:digit:]]")))
+  }
+  breaks <- as.numeric(formatC(0 + breaks, digits = avgDigits, width = 1L))
+
+  # If now, after rounded, our max bin does not include the max of the data, add another bin
+  if (max(breaks) < max_x) {
```

Comment:
I think this might need to be `<= max_x` ?
I'm seeing some extra empty bins now via the EDA.

This is SCORE Mozambique. The extra bin is empty, and no data seems to be missing from the preceding 8 bins. I'll see if I can change this and run the tests. I'm not sure I can make a new test though.

![image](https://user-images.githubusercontent.com/308639/174081256-40d350df-dc24-4dd5-852c-5e93ad0552f1.png)